components/places/android/src/main/java/mozilla/appservices/places/PlacesConnection.kt

```diff
@@ -313,6 +313,24 @@ open class PlacesReaderConnection internal constructor(connHandle: Long) :
         }
     }
 
+    override fun getVisitPageWithBound(
+        offset: Long,
+        bound: Long,
+        count: Long,
+        excludeTypes: List<VisitType>
+    ): List<VisitInfo> {
+        val infoBuffer = rustCall { error ->
+            LibPlacesFFI.INSTANCE.places_get_visit_page_with_bound(
+                    this.handle.get(), offset, bound, count, visitTransitionSet(excludeTypes), error)
+        }
+        try {
+            val infos = MsgTypes.HistoryVisitInfos.parseFrom(infoBuffer.asCodedInputStream()!!)
+            return VisitInfo.fromMessage(infos)
+        } finally {
+            LibPlacesFFI.INSTANCE.places_destroy_bytebuffer(infoBuffer)
+        }
+    }
```

Comment:
As a caller of this function, I would expect to be calling it in order to do some sort of traversal of multiple pages. How do I know what the right value of `bound` is to pass in?  For example, how do I write this loop:

```
bound = offset = 0
while (more pages) {
  page = getVisitPageWithBound(bound, offset, N, EXCLUDE)
  // do something with page
  bound = ??? // what's the next value of bound?
  offset = ???  // what's the next value of offset?
}
```

I expect that the caller will need to calculate them from the list of `VisitInfo`s that was returned, but we need to provide some instructions for how (or somehow return it from the previous call).

Reply:
Yes. I agree. And maybe return the next `bound` and `offset` can be even nicer than providing instructions?
So, if new bound is the same as previous bound, return the same `bound` with `offset` plus the size of page; If new bound is a different bound, return new `bound` with `offset=0`.
Does this sound right?

Reply:
Yes, that sounds right. I agree that returning it somehow would be better, but it's not very obvious to me how to return it in a nice way over the FFI.

Reply:
I am thinking of creating a protobuf message type `OffsetToken` which contains the next `bound` and `offset` (both i64).